### PR TITLE
Paint by county with ID instead of features

### PR DESCRIPTION
**Pull request by nofurtherinformation**
Instead of painting by county using features, this PRs drafts a method of caching all IDs and then filtering based on those

## Description
- Cache IDs of features on `data` in the map
- When painting by county, search those IDs

## Reviewers
- Primary: @raphaellaude 
- Secondary: @mariogiampieri

### CODE REVIEW

**Comment by nofurtherinformation** [comment]
Also filters counties to only active state

File: app/src/app/store/mapRenderSubs.ts [hunk 1]
```diff
@@ -303,6 +311,18 @@ export const getRenderSubscriptions = (useMapStore: typeof _useMapStore) => {
...
   );
+  
+  const filterCountiesSub = useMapStore.subscribe<[string|undefined, MapStore['getMapRef']]>(state => [state.mapOptions.currentStateFp, state.getMapRef],
+    ([stateFp, getMapRef]) => {
+      const mapRef = getMapRef()
+      if (!mapRef) return
+      const filterExpression = (stateFp ? ["==", "STATEFP", stateFp] : true) as any
+      COUNTY_LAYERS.forEach(layer => {
+        mapRef.setFilter(layer,  ["any", filterExpression])
+      })
```
Comment (raphaellaude) [inline app/src/app/store/mapRenderSubs.ts:322]: H: Need to check that the county layer is added first

image

File: app/src/app/utils/helpers.ts [hunk 2]
```diff
@@ -133,16 +134,25 @@ export const getFeaturesIntersectingCounties = (
...
-  });
-
-  return filterFeatures(
-    features,
-    true,
-    [(feature) => Boolean(feature?.id && feature.id.toString().match(/\d{5}/)?.[0] === fips)]
-  );
+  return filterFeatures([...cachedParentFeatures, ...childFeatures], true, [
+    feature => Boolean(feature?.id && feature.id.toString().match(/\d{5}/)?.[0] === fips),
+  ]);
```
Comment (raphaellaude) [inline app/src/app/utils/helpers.ts:155]: Random comment but we should probably start testing the FE, esp. would be easy to add unit tests for little utilities like these even if we don't go full FE testing w/ a framework

**Review by raphaellaude — CHANGES_REQUESTED** [review]
Sort of rushed through this review so apologies it's not more in depth. Only one comment re: unavailable county layers breaking things

File: app/src/app/store/mapRenderSubs.ts (hunk 1 above)
Comment (nofurtherinformation) [reply app/src/app/store/mapRenderSubs.ts:322]: Ah yep

File: app/src/app/utils/helpers.ts (hunk 2 above)
Comment (raphaellaude) [reply app/src/app/utils/helpers.ts:155]: Here's an issue for that #208

File: app/src/app/store/mapRenderSubs.ts [hunk 3]
```diff
@@ -308,6 +316,18 @@ export const getRenderSubscriptions = (useMapStore: typeof _useMapStore) => {
...
   );
+  
+  const filterCountiesSub = useMapStore.subscribe<[string|undefined, MapStore['getMapRef']]>(state => [state.mapOptions.currentStateFp, state.getMapRef],
+    ([stateFp, getMapRef]) => {
+      const mapRef = getMapRef()
+      if (!mapRef) return
+      const filterExpression = (stateFp ? ["==", "STATEFP", stateFp] : true) as any
+      COUNTY_LAYERS.forEach(layer => {
+        mapRef.getLayer(layer) && mapRef.setFilter(layer,  ["any", filterExpression])
+      })
```
Comment (raphaellaude) [inline app/src/app/store/mapRenderSubs.ts:327]: I: This doesn't seem to be capturing the point label id

image

**Review by raphaellaude — CHANGES_REQUESTED** [review]
almost there!

File: app/src/app/store/mapRenderSubs.ts (hunk 3 above)
Comment (nofurtherinformation) [reply app/src/app/store/mapRenderSubs.ts:327]: Had a typo in the layer name but should be good now

**Review by raphaellaude — APPROVED** [review]
Lgtm

**Pull request merged**